src/mds/objects.py

```diff
@@ -226,6 +226,73 @@ async def create_object_for_id(
     return JSONResponse(response, HTTP_201_CREATED)
 
 
+@mod.get("/objects")
+async def get_objects(
+    request: Request,
+    #  XXX would be nice to be able to specify whether to return indexd records
+    #  (e.g GET objects?data=metadata would only return mds objects)
+    data: bool = Query(
+        False,
+        description="Switch to returning a list of GUIDs (false), "
+        "or GUIDs mapping to their metadata (true).",
+    ),
+    limit: int = Query(
+        10, description="Maximum number of records returned. (max: 2000)"
+    ),
+    offset: int = Query(0, description="Return results at this given offset."),
+    #  XXX description
+    #  XXX how to name this python variable something other than filter but
+    #  still have client use "filter" as URL query param? (bc filter is already
+    #  built in to Python)
+    filter: str = Query("", description="Filters to apply."),
+) -> JSONResponse:
+    """
+    XXX comments
+    """
+
+    metadata_objects = await search_metadata_helper(
+        data=data, limit=limit, offset=offset, filter=filter
+    )
+
+    records = {}
+    if metadata_objects:
+        try:
+            endpoint_path = "/bulk/documents"
+            full_endpoint = config.INDEXING_SERVICE_ENDPOINT.rstrip("/") + endpoint_path
+            guids = (
+                list(metadata_objects.keys())
+                if hasattr(metadata_objects, "keys")
+                else metadata_objects
```

Comment:
what exactly is this `if` checking? should we just use `if data`?

Reply:
Good call. Let's use `if data` here. `if` was just checking if `search_metadata_helper` returned a list of guids or a dictionary of guids to objects.